htdocs/core/class/commonobject.class.php

```diff
@@ -1208,12 +1208,15 @@ public function delete_linked_contact($source = '', $code = '')
 	{
 		// phpcs:enable
 		$temp = array();
+		$listId = null;
 		$typeContact = $this->liste_type_contact($source, '', 0, 0, $code);
 
-		foreach ($typeContact as $key => $value) {
-			array_push($temp, $key);
+		if (! empty($typeContact)) {
+			foreach ($typeContact as $key => $value) {
+				array_push($temp, $key);
+			}
+			$listId = implode(",", $temp);
```

Comment:
Something looks strange to me in this method.
We we don't not find any type of contact, typeContact will be empty. It means we will delete all links later with the delete ?
It seems it was already the case before. 
Don't you think we should include the delete inside the
 if ($listId)